docs/tracing.md

```diff
@@ -0,0 +1,36 @@
+# Tracing
+
+Cortex uses [Jaeger](https://www.jaegertracing.io/) to implement distributed
+tracing. We have found Jaeger invaluable for troubleshooting the behavior of
+Cortex in production.
+
+## Dependencies
+
+In order to send traces you will need to set up a Jaeger deployment. A
+deployment includes either the jaeger all-in-one binary, or else a distributed
+system of agents, collectors, and queriers.  If running on Kubernetes, [Jaeger
+Kubernetes](https://github.com/jaegertracing/jaeger-kubernetes) is an excellent
+resource.
+
+## Configuration
+
+In order to configure Cortex to send traces you must do two things:
+1. Set the `JAEGER_AGENT_HOST` environment variable to point to your Jaeger
+   agent. This defaults to `localhost`.
+1. Enable sampling somewhere. The general options for this are:
+   * Upstream of Cortex, perhaps in your frontdoor.
+   * In Cortex itself.  
+
+Note that you must specify one of `JAEGER_AGENT_HOST` or
+`JAEGER_SAMPLER_MANAGER_HOST_PORT` in each component for Jaeger to be enabled,
+even if you plan to use the default values.
+
+If you would like to enable sampling in Cortex you can specify either
+`JAEGER_SAMPLER_MANAGER_HOST_PORT` for remote sampling, or
+`JAEGER_SAMPLER_TYPE` and `JAEGER_SAMPLER_PARAM` to manually set sampling
+configuration. You will want to specify the environment variables in the
+query-frontend, distributor, and ruler since those are where requests start.
```

Comment:
We also sample in ingester, to catch the writes to DB which are self-initiated

Reply:
Good call, thanks!